core/drivers/gpio/gpio.c

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2022, Microchip
+ */
+
+#include <drivers/gpio.h>
+#include <libfdt.h>
+#include <stdio.h>
+#include <tee_api_defines.h>
+#include <tee_api_defines_extensions.h>
+#include <tee_api_types.h>
+#include <util.h>
+
+struct gpio *gpio_get_dt_two_cells(struct dt_driver_phandle_args *a,
+				   TEE_Result *res)
+{
+	struct gpio *gpio = NULL;
+
+	if (a->args_count != 2) {
+		*res = TEE_ERROR_BAD_PARAMETERS;
```

Comment:
I would even add an assertion here since by DT bindings we expect 2 cells data. (not a strong opion yet)